src/osd/ReplicatedPG.cc

```diff
@@ -1657,14 +1657,13 @@ void ReplicatedPG::do_op(OpRequestRef& op)
   // op to succeed despite being full).  The except is FULL_FORCE|FULL_TRY ops,
   // which there is no reason to discard because they bypass all full
   // checks anyway.
-  // FIXME: we exclude mds writes for now.
-  if (!(m->get_source().is_mds() || m->has_flag(CEPH_OSD_FLAG_FULL_FORCE|CEPH_OSD_FLAG_FULL_TRY)) &&
+  if (!(m->has_flag(CEPH_OSD_FLAG_FULL_FORCE|CEPH_OSD_FLAG_FULL_TRY)) &&
       info.history.last_epoch_marked_full > m->get_map_epoch()) {
     dout(10) << __func__ << " discarding op sent before full " << m << " "
 	     << *m << dendl;
     return;
   }
-  if (!(m->get_source().is_mds() || m->has_flag(CEPH_OSD_FLAG_FULL_TRY)) &&
+  if (!(m->has_flag(CEPH_OSD_FLAG_FULL_TRY)) &&
       osd->check_failsafe_full()) {
```

Comment:
this line still needs to be fixed.  the client should not be able to bypass the failsafe check.